proto/interchain_security/ccv/v1/ccv.proto

```diff
@@ -58,17 +58,19 @@ message MaturedUnbondingOps {
   repeated uint64 ids = 1;
 }
 
-// ConsumerPacketData contains a consumer packet data and a type tag
+// ConsumerPacketData contains a consumer packet data, type tag, and index for storage.
 message ConsumerPacketData {
   ConsumerPacketDataType type = 1; 
 
   oneof data {
     SlashPacketData slashPacketData = 2;
     VSCMaturedPacketData vscMaturedPacketData = 3;
   }
+  uint64 idx = 4;
```

Comment:
Extra care should be taken in reviewing this PR, as it does mutate a proto file. Note this index field was added to the `ConsumerPacketData` type so that we can use that index in `PendingDataPacketsKey`, to implement a queue with constant append time. See `AppendPendingPacket`